src/hooks/use-load-data-pages.ts

```diff
@@ -23,9 +23,33 @@ const shouldFetchAdditionalData = (
   );
 };
 
+const isMissingLayoutData = (layoutService: LayoutService): boolean => {
```

Comment:
I'm thinking this could be simplified into:

```js
const isMissingLayoutData = (layoutService: LayoutService): boolean => {
  const { qHyperCube } = layoutService.layout;
  const { qWidth, qHeight } = qHyperCube.qPivotDataPages[0]?.qArea ?? { qWidth: 0, qHeight: 0 };

  return qWidth < DEFAULT_PAGE_SIZE || qHeight < DEFAULT_PAGE_SIZE;
};
```